web/modules/custom/ai_screening_project_track/src/Helper/ProjectTrackHelper.php

```diff
@@ -152,10 +152,83 @@ public function deleteProjectTracks(array $projectTracks): void {
    * @throws \Drupal\Core\Entity\EntityStorageException
    */
   public function projectTrackToolComputed(ProjectTrackToolComputedEvent $event): void {
-    $tool = $event->getTool();
-    $track = $tool->getProjectTrack();
+    $track = $event->getTool()->getProjectTrack();
     $trackConfig = $track->getConfiguration();
-    $tools = $this->getToolsData($track);
+    $trackConfig['bubbleChartReportResult'] = $trackConfig['bubbleChartReportResult'] ?? [];
+    $trackConfig['submissionReportResult'] = $trackConfig['submissionReportResult'] ?? [];
+    $toolsData = $this->getToolsData($track);
+    $configResult = [];
+
+    $reportTypeValues = $track->getType()->get('field_report_type')->getValue();
+    $reportTypes = array_map(function ($reportTypeValues) {
+        return $reportTypeValues['value'];
+    }, $reportTypeValues);
+
+    if (in_array('bubble_chart', $reportTypes)) {
+      $configResult['bubbleChartReportResult'] = $this->computeBubbleChartReportValues($toolsData, $track, $trackConfig['bubbleChartReportResult']);
+    }
+
+    if (in_array('webform_submission', $reportTypes)) {
```

Comment:
Would it make sense to use `elseif` here to make it clear that we only want to handle one of the cases (or to signal that they are mutually exclusive):

```suggestion
    } elseif (in_array('webform_submission', $reportTypes)) {
```